modules/node_groups/launch_template.tf

```diff
@@ -63,6 +63,12 @@ resource "aws_launch_template" "workers" {
     ])
   }
 
+  metadata_options {
+    http_endpoint               = lookup(each.value, "metadata_http_endpoint", "enabled")
```

Comment:
Let's generate `metadata_options` using `dynamic` and create it only if one of 3 inputs are present. Exactly as here - https://github.com/terraform-aws-modules/terraform-aws-autoscaling/blob/master/main.tf#L92-L99